### DISCUSSION TITLE
Start wallpaper video at a random point

### DISCUSSION THREAD

**Discussion by deselrahc** [rocksdanister/lively#1823] [Ideas]
Is it possible to have a random 'button' with each wallpaper file to get it to start at a random time point by Lively? I have a number of wallpapers that are 'dynamic patterns' and they always start at the same place when I start Lively. It would be great if I could toggle a random 'button' to allow Lively to start the wallpaper at a random point. For example I have a YouTube video that is 1 hour in length, and having it always start at the same point in time is boring. If it were to start randomly at 10mins in, 6mins in, 9½mins in, etc. every time Lively starts, that would be great. Not sure how this would work with all the wallpapers out there, but it would be a good idea, imho!!
deselrahc

**Comment by rocksdanister** [comment]
Script can be written using lively commandline control (video files only):
https://github.com/rocksdanister/lively/wiki/Command-Line-Controls#seek-wallpaper

**Reply by deselrahc** [reply]
Thnx @rocksdanister but I need a little more help! From what I can see, I can set the seek time to an absolute percentage or a relative percentage. This would certainly start the video file at the chosen time mark, but (if I understand correctly) the next time Lively starts (usually the next day for me) the video would again start at the previously set time mark (percentage). So, yes the playback won't start at the beginning, but the new start time mark will be the same again, and again, every time I start Lively. Is there a way to make the selection of the start time mark truly random?
TIA
deselrach

**Reply by deselrahc** [reply]
Sorry for multiple posts @rocksdanister... just to clarify. If I use the relative position:

Relative: Percentage(-100 to 100) value starting with + or -, seeks from the current media position.
Example: seekwp --value +10

Isn't the 'current media position (for my video wallpaper) going t o be the start of the video when I start Lively each time I start my laptop?
TIA (again)
deselrahc

**Reply by rocksdanister** [reply]
No need to ping me, I get notification regardless.

Yes current will always be from beginning, you will have to write a python/ahk script with random number generation:
https://github.com/rocksdanister/lively/wiki/Command-Line-Controls#python

and then run it as schedule task with some delay (for lively to start.)
https://www.youtube.com/watch?v=iEb5DVMz3Ds

**Reply by deselrahc** [reply]
OMG! Okay... not in my 'pay grade' so to speak. I haven't written scripts (ever) and my coding experience is circa early Pascal/Prolog. So I think I'll leave this alone. Thnx for all your help! I love Lively heaps.
Cheers
deselrahc
